config/impl.go

```diff
@@ -21,54 +21,88 @@
 package config
 
 import (
+	"context"
 	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+
 	"go.ketch.com/lib/orlop/v2/env"
 	"go.ketch.com/lib/orlop/v2/errors"
 	"go.ketch.com/lib/orlop/v2/service"
-	"go.uber.org/fx"
-	"reflect"
-	"strings"
 )
 
-type Config interface {
-	Options() fx.Option
-}
-
 type providerImpl struct {
+	configs map[string]any
 	environ env.Environ
+	prefix  service.Name
 }
 
-func New(environ env.Environ) Provider {
+func New(environ env.Environ, prefix service.Name, p Params) Provider {
```

Comment:
If you have `Params`, why not move all of the required Params there instead of leaving `environ` and `prefix` here?